drivers/timer/riscv_machine_timer.c

```diff
@@ -10,7 +10,7 @@
 
 #define CYC_PER_TICK ((uint32_t)((uint64_t)sys_clock_hw_cycles_per_sec()	\
 			      / (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC))
-#define MAX_CYC 0xffffffffu
+#define MAX_CYC 0x7fffffffu
```

Comment:
FWIW, as I read the code this was a signedness bug.  The tick argument to z_clock_set_timeout() is an int32_t, so the old value was -1.  I'd just use INT_MAX here, and let it be a signed value for clarity.  The "u" suffix isn't actually doing anything useful.

Reply:
Thanks for review. Indeed, I fixed it to INT_MAX.